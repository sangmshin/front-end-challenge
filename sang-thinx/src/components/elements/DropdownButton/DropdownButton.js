import React from 'react';
import { func, string } from 'prop-types';
import { Button } from 'components/elements';
import styles from './DropdownButton.module.scss';

export const DropdownButton = ({
  onClick,
  selectedItem,
  type,
}) => {
  const buttonLabel = selectedItem ? selectedItem : type;

  return (
    <Button
      className={styles.root}
      onClick={onClick}
      role='button'
    >
      {buttonLabel}
      <span className={styles.caret}></span>
    </Button>
  )
}

DropdownButton.propTypes = {
  onClick: func.isRequired,
  selectedItem: string,
  type: string.isRequired,
}

DropdownButton.defaultProps = {
  selectedItem: null
}

export default DropdownButton;