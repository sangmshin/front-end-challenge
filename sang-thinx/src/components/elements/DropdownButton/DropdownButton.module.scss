.root {
  display: flex;
  justify-content: space-between;
  align-items: stretch;
  width: 100%;

  .caret {
    position: relative;
    cursor: pointer;
    transition: .3s ease-out;

    &.open {
      transform: rotateX(180deg);
    }
    
    &:before {
      content: '';
      position: absolute;
      top: 6px;
      right: 7px;
      border-top: 6px solid #000;
      border-left: 6px solid transparent;
      border-right: 6px solid transparent;
    }
    
    &:after {
      content: '';
      position: absolute;
      top: 3px;
      right: 7px;
      border-top: 6px solid #f8f8f8;
      border-left: 6px solid transparent;
      border-right: 6px solid transparent;
    }
  }
}
