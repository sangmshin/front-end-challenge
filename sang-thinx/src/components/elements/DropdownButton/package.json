{
  "name": "DropdownButton",
  "main": "./DropdownButton.js"
}
