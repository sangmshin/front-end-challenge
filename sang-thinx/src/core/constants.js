export const COLOR_HEXCODE_MAP = {
  beige: '#f2d9a5',
  black: '#000',
}