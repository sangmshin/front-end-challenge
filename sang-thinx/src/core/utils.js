const SMALL = 768; /* Small screen / tablet */
const MEDIUM = 1024; /* Medium screen / desktop */
const LARGE = 1200; /* Large screen / wide desktop */
let BREAKPOINT;

/**
 * Finds the current matched media query breakpoint
 *
 * @returns {String} matched media query breakpoint [large, medium, small, xsmall]
 */
export function getCurrentBreakPoint() {
  if (!BREAKPOINT) {
    BREAKPOINT = {
      large: window.matchMedia(`(min-width: ${LARGE}px)`),
      medium: window.matchMedia(`(min-width: ${MEDIUM}px)`),
      small: window.matchMedia(`(min-width: ${SMALL}px)`),
      xsmall: window.matchMedia(`(max-width: ${SMALL - 1}px)`)
    };
  }

  return Object.keys(BREAKPOINT).find(breakpoint => BREAKPOINT[breakpoint].matches);
}

/**
 * Determines if the data is array and has at least one or more items
 * @return {boolean}
 */
export const isArrayWithLength = arr => Array.isArray(arr) && arr.length > 0;

/**
 * A function with no operation. (for testing)
 */
export const noop = () => null;
