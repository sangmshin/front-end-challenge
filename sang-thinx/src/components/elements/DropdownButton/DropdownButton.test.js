import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DropdownButton } from './DropdownButton';
import { noop } from 'core/utils';

const wrapper = shallow(
  <DropdownButton onClick={noop} type='age' selectedItem='xsmall' />
)

describe('Component :: DropdownButton', () => {
  describe('#render', () => {
    it('should match snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
