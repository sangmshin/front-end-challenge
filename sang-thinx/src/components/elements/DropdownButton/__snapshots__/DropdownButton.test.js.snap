// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component :: DropdownButton #render should match snapshot 1`] = `
<ForwardRef
  accessibleText=""
  ariaLabel=""
  className="root"
  disabled={false}
  onClick={[Function]}
  role="button"
  style={null}
>
  xsmall
  <span
    className="caret"
  />
</ForwardRef>
`;
